modules/tensor_mechanics/include/materials/RadialReturnStressUpdate.h

```diff
@@ -28,26 +28,35 @@ InputParameters validParams<RadialReturnStressUpdate>();
  * Petrinic's Introduction to Computational Plasticity (2004) Oxford University Press.
  */
 
-class RadialReturnStressUpdate : public StressUpdateBase
+class RadialReturnStressUpdate : public StateUpdateBase
 {
 public:
   RadialReturnStressUpdate(const InputParameters & parameters);
 
   /// A radial return (J2) mapping method is defined in this material by overwritting
   /// the computeInelasticStrainIncrement method.
-  virtual void updateStress(RankTwoTensor & strain_increment,
-                            RankTwoTensor & inelastic_strain_increment,
-                            RankTwoTensor & stress_new) override;
+  virtual void updateState(RankTwoTensor & strain_increment,
+                           RankTwoTensor & inelastic_strain_increment,
+                           const RankTwoTensor & rotation_increment,
+                           RankTwoTensor & stress_new,
+                           const RankTwoTensor & stress_old,
+                           const RankFourTensor & elasticity_tensor,
+                           const RankTwoTensor & elastic_strain_old,
+                           bool compute_full_tangent_operator,
+                           RankFourTensor & tangent_operator) override;
```

Comment:
I'm glad to have access to all of these properties for some more complex crystal plasticity models with which I'm working, but I'm a bit concerned about the overhead for the isotropic classes--is it only the beam rotation test that slows down?